src/table/internal.tsx

```diff
@@ -117,7 +118,13 @@ const InternalTable = React.forwardRef(
 
     const { moveFocusDown, moveFocusUp, moveFocus } = useFocusMove(selectionType, items.length);
     const { onRowClickHandler, onRowContextMenuHandler } = useRowEvents({ onRowClick, onRowContextMenu });
-    const visibleColumnDefinitions = visibleColumns
+
+    const sortedVisibleColumnDefinitions = columnDisplay
```

Comment:
This part of code is difficult to read and not not optimal as well. I'd recommend creating a helper function like `getVisibleColumnDefinitions`:

```
function getVisibleColumnDefinitions({ columnDisplay, visibleColumns, columnDefinitions }) {
  // columnsDisplay has a precedence over deprecated visibleColumns.
  if (columnDisplay) {
    return getVisibleColumnDefinitionsFromColumnDisplay({ columnDisplay, columnDefinitions })
  } else if (visibleColumns) {
    return getVisibleColumnDefinitionsFromVisibleColumns({ visibleColumns, columnDefinitions })
  } else {
    return columnDefinitions;
  }
}

function getVisibleColumnDefinitionsFromColumnDisplay({ columnDisplay, columnDefinitions }) {
   ...
}

function getVisibleColumnDefinitionsFromVisibleColumns({ visibleColumns, columnDefinitions }) {
   ...
}
```

Reply:
Addressed in [58f02ff](https://github.com/cloudscape-design/components/pull/804/commits/58f02ff5c815612fd8623d9239243a20b32d2215).